components/Tabs/Tabs.js

```diff
@@ -1,38 +1,40 @@
 class TabLink {
   constructor(element){
     // assign this.element to the element reference
-    this.element;
+    this.element = element;
     // Get the tab data attribute and save the value here
-    this.tabData = ; 
+    this.tabData = this.element.dataset.tab; 
     // Find all elements with the .card class in index.html that correspond to the tab data attribute
     // If the data is 'all' then select all cards regardless of their data attribute
-    if(){
-      this.cards = ;
+    if(`${this.tabData}` === 'all'){
+      this.cards = document.querySelectorAll('.card');
     } else {
-      this.cards = ;
+      this.cards = document.querySelectorAll(`.card[data-tab="${this.tabData}"]`);
     }
 
     // Map over the cards array and convert each card element into a new instance of the TabCard class. 
     // Pass in the card object to the TabCard class.
-    this.cards = Array.from(this.cards).map();
+    this.cards = Array.from(this.cards).map(card => new TabCard(card));
     // Add a click event that invokes selectTab
-    this.element.addEventListener();
+    this.element.addEventListener("click", () => this.selectTab());
+
   }
 
   selectTab(){
 
     // Select all elements with the .tab class on them
-    const tabs = document.querySelectorAll();
+    const tabs = document.querySelectorAll('.tab');
     // Iterate through the NodeList removing the .active-tab class from each element
-    tabs.forEach()
+    tabs.forEach(tab => tab.classList.remove('active-tab'));
     // Add a class of ".active-tab" to this.element
-    this.element;
+    this.element.classList.add('active-tab');
 
 
     // Select all of the elements with the .card class on them
-    const cards = ;
+    const cards = document.querySelectorAll('.card');
     // Iterate through the NodeList setting the display style each one to 'none'
-    cards.forEach()
+    // Not sure why this works and card.style.display= "none" does not.
+    cards.forEach(card => card.style = 'display:none');
```

Comment:
what's the difference between `card.style = 'display:none'` 🆚 `card.style.display = 'none'`?